DynamicMeter.c

```diff
@@ -97,17 +98,20 @@ static const char* DynamicMeter_getCaption(const Meter* this) {
 }
 
 static void DynamicMeter_getUiName(const Meter* this, char* name, size_t length) {
+   assert(length > 0);
+
    const Settings* settings = this->host->settings;
    const DynamicMeter* meter = Hashtable_get(settings->dynamicMeters, this->param);
    if (meter) {
       const char* uiName = meter->caption;
       if (uiName) {
-         int len = strlen(uiName);
+         size_t len = strlen(uiName);
          if (len > 2 && uiName[len - 2] == ':')
             len -= 2;
-         xSnprintf(name, length, "%.*s", len, uiName);
+
+         String_safeStrncpy(name, uiName, MINIMUM(length, len + 1));
```

Comment:
The final `+1` doesn't seem necessary as `String_safeStrncpy` enforces NUL-termination of the destination buffer.

Reply:
Also, for `len==0` (AKA empty caption) this is UB.

Reply:
@BenBE The length in `String_safeStrncpy` includes the terminating NUL byte while `strlen` return value does not include it. Hence, the `+1` is intentional and required.

And yes, `len==0` means you get an empty string copied (i.e. just one NUL byte). No UB.

Reply:
Confusion on my side. I misread the implementation of `String_safeStrncpy` regarding the termination for how much to copy from the source buffer, thus the "no need for the +1"; the `len==0` was a follow-up on that mistake. My bad.